### PR TITLE
Feature/vizbox

### PR DESCRIPTION
**Pull request by LoyVanBeek**

### CODE REVIEW

File: launch/interaction/vizbox.launch [hunk 1]
```diff
@@ -6,13 +6,15 @@
...
-    <node pkg="vizbox" type="server.py" name="vizbox" cwd="node" output="screen">
+    <node pkg="vizbox" type="server.py" name="vizbox" cwd="node" output="log">
         <remap to="text_to_speech/output" from="robot_text"/>
         <remap to="hmi/result/sentence" from="operator_text"/>
+        <remap to="top_kinect/rgb/image" from="image"/>
     </node>
 
     <node pkg="topic_tools" type="transform" name="extract_hmi_result_sentence"
-        args="hmi/result hmi/result/sentence std_msgs/String 'm.result.sentence'"/>
+        args="/$(arg robot_name)/hmi/result hmi/result/sentence std_msgs/String 'm.result.sentence'"/>
```
Comment (MatthijsBurgh) [inline launch/interaction/vizbox.launch:20]: Or include this launch file in the robot ns?

Comment (LoyVanBeek) [reply launch/interaction/vizbox.launch:20]: It is, but topic_tools doesn't handle namespaces AFAICT. 
hmi/result is in the same namepace and the ROS namespacing works for that.

Comment (LoyVanBeek) [reply launch/interaction/vizbox.launch:20]: Of course I tried without ```/$(arg robot_name)/```, but that results in an error:

```
ERROR: Wrong input topic (or topic field): hmi/result
[amigo/extract_hmi_result_sentence-18] process has died [pid 12326, exit code 1, cmd /opt/ros/kinetic/lib/topic_tools/transform hmi/result hmi/result/sentence std_msgs/String m.result.sentence __name:=extract_hmi_result_sentence __log:=/home/loy/.ros/log/7310f27a-4d6e-11e8-bcb2-d07e35c47b06/amigo-extract_hmi_result_sentence-18.log].
log file: /home/loy/.ros/log/7310f27a-4d6e-11e8-bcb2-d07e35c47b06/amigo-extract_hmi_result_sentence-18*.log
```

Comment (MatthijsBurgh) [reply launch/interaction/vizbox.launch:20]: 👍

File: launch/interaction/vizbox.launch [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+
+<launch>
+
+    <!-- Get the machine file -->
+    <arg name="machine" default="localhost"/>
```
Comment (jlunenburg) [inline launch/interaction/vizbox.launch:6]: I guess machine files aren't used anymore?!

File: launch/interaction/vizbox.launch [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    <include file="$(env ROBOT_BRINGUP_PATH)/machines/$(arg machine).machine" />
+
+    <arg name="robot_name"/>
+
+    <node pkg="vizbox" type="server.py" name="vizbox" cwd="node" output="log">
+        <remap to="text_to_speech/output" from="robot_text"/>
```
Comment (jlunenburg) [inline launch/interaction/vizbox.launch:14]: Being picky, isn't it easier to read from .. to?

File: launch/interaction/vizbox.launch [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+<launch>
+
+    <!-- Get the machine file -->
+    <arg name="machine" default="localhost"/>
+    <include file="$(env ROBOT_BRINGUP_PATH)/machines/$(arg machine).machine" />
+
+    <arg name="robot_name"/>
+
+    <node pkg="vizbox" type="server.py" name="vizbox" cwd="node" output="log">
```
Comment (jlunenburg) [inline launch/interaction/vizbox.launch]: Is vizbox installed by default?

**Review by jlunenburg — CHANGES_REQUESTED** [review]
Have a look at the comments. If you don't use a machine files, don't include them? Furthermore, did you test this on the robot?

File: launch/interaction/vizbox.launch (hunk 4 above)
Comment (LoyVanBeek) [reply launch/interaction/vizbox.launch]: https://github.com/tue-robotics/tue-env/pull/261

File: launch/interaction/vizbox.launch (hunk 2 above)
Comment (LoyVanBeek) [reply launch/interaction/vizbox.launch:6]: Yes they are, all the state machine launches declare which machine to run each node in. I just forgot to pass that on to the actual node

**Comment by MatthijsBurgh** [comment]
@LoyVanBeek please decide what to with the vizbox.

**Comment by LoyVanBeek** [comment]
I think it's handy for us to have regards of RoboCup, so I'd say merge it ASAP

**Comment by LoyVanBeek** [comment]
Nobody cares

File: launch/interaction/vizbox.launch [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+
+<launch>
+
+    <!-- Get the machine file -->
+    <arg name="machine" default="localhost"/>
+    <arg name="image" default="top_kinect/rgb/image"/>
+    <include file="$(env ROBOT_BRINGUP_PATH)/machines/$(arg machine).machine" />
```
Comment (MatthijsBurgh) [inline launch/interaction/vizbox.launch]: better to put this `include` directly after the machine argument.

File: launch/interaction/vizbox.launch (hunk 1 above)
Comment (MatthijsBurgh) [reply launch/interaction/vizbox.launch:20]: But does it output to `/robot_name/hmi/result/sentence`? Or to `hmi/result/sentence`? What is the desired behaviour? It is without the robot name. Please make it absolute.

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

**Pull request merged**